### PR TITLE
Refactor alerts

### PR DESCRIPTION
**Pull request by hahnbeelee**

### CODE REVIEW

File: server/src/helpers/github/types.ts [hunk 1]
```diff
@@ -17,6 +17,8 @@ export type Alert = {
     filename: string;
     lineRange: LineRange;
     type: string;
+    code: CodeType;
+    githubCommentId?: string;
```
Comment (hahnbeelee) [inline server/src/helpers/github/types.ts:21]: unique identifier for the github comment

File: server/src/routes/tasks.ts [hunk 2]
```diff
@@ -50,6 +53,13 @@ tasksRouter.post('/update/:docId', userMiddleware, async (req, res) => {
     type: 'update',
     url: doc.url,
   });
+
+  track(res.locals.user.userId, 'Task Created', {
+    id: task._id.toString(),
+    org: org._id,
+    doc: docId,
+    type: 'update'
+  })
```
Comment (hahnbeelee) [inline server/src/routes/tasks.ts:62]: Specifically can u look at if this tracking looks good?

Comment (hanywang2) [reply server/src/routes/tasks.ts:62]: You gotta do org._id.toString(), and docId doesn't need to be converted into a type object. It should stay as a string

File: server/src/routes/tasks.ts [hunk 3]
```diff
@@ -58,7 +68,69 @@ tasksRouter.delete('/:taskId', userMiddleware, async (req, res) => {
...
+  if (task != null && task.status !== status && status === 'done') { // task is resolved
+    try {
+      // FindOne might cause an issue with separate installations on the same org
+      const org = await Org.findOne({'integrations.github.installations': {
+        $elemMatch: {
+            'account.login': gitOrg
+        }
+      }});
+      const formattedOrg = removeUnneededDataFromOrg(org);
+      track(formattedOrg._id, 'Task Completed', task);
```
Comment (hanywang2) [inline server/src/routes/tasks.ts:127]: ditto

File: server/src/routes/tasks.ts [hunk 4]
```diff
@@ -58,7 +68,69 @@ tasksRouter.delete('/:taskId', userMiddleware, async (req, res) => {
...
+      type: 'review',
+      source: 'github',
+      url: alert.url,
+      githubCommentId: alert?.githubCommentId
+    };
+    return task;
+  });
+  const tasksResponse = await Task.insertMany(tasks);
+  const trackPromises = tasksResponse.map((task:any) => {
+    return track(org.toString(), 'Task Created', task);
```
Comment (hanywang2) [inline server/src/routes/tasks.ts:98]: Set `isOrg: true` as a field in the data, if you are tracking an org

**Review by hanywang2 — APPROVED** [review]

**Pull request merged**